syfertext/tokenizer.py

```diff
@@ -175,18 +217,411 @@ def __call__(self, text: Union[String, str]):
+
+        # If subtring is remaining after splitting all the affixes.
+        if substring:
+            # Create the TokenMeta object
+            end_pos = start_pos + len(substring) - 1
+            token_meta = TokenMeta(
+                hash_key=self.vocab.store[(substring)],
+                start_pos=start_pos,
+                end_pos=end_pos,
+                space_after=False,  # for the last token space_after will be updated explicitly according to the original substring.
+                is_space=False,
+            )
+
+            # Append the token to the document
+            doc.container.append(token_meta)
+
+        # Append the infixes TokenMeta list in doc
+        doc.container.extend(affixes["infix"])
+
+        # Append the suffixes TokenMeta list in doc
+        doc.container.extend(reversed(affixes["suffix"]))
+
+        # Get the last token and update it's space_after attr according to original substring's TokenMeta data
+        doc.container[-1].space_after = space_after
+
+        return doc
+
+    def _get_prefix_token_meta(self, substring: str, pos: int) -> Tuple[TokenMeta, str, int]:
+        """Makes TokenMeta data for substring which are prefixes.
+
+        Args:
+            substring: The substring to tokenize.
+            pos: The pointer to the start position of substring in the text.
+
+        Returns:
+            token_meta: The TokenMeta object with TokenMeta data of prefix.
+            substring: The updated substring after removing prefix.
+            pos: The pointer to the start position of new substring in the text.
+        """
+        pre_len = self.find_prefix(substring)
+        # break if pattern matches the empty string
+        if pre_len == 0:
+            return None, substring, pos
+        end_pos = pos + pre_len - 1
+
+        # Create the TokenMeta object
+        token_meta = TokenMeta(
+            hash_key=self.vocab.store[str(substring[:pre_len])],
+            start_pos=pos,
+            end_pos=end_pos,
+            space_after=False,  # for the last token space_after will be updated explicitly according to the original substring.
+            is_space=False,
+        )
+
+        pos = end_pos + 1
+
+        # Update remaining substring after removing the prefix.
+        substring = substring[pre_len:]
+
+        return token_meta, substring, pos
+
+    def _get_suffix_token_meta(self, substring: str, pos: int) -> Tuple[TokenMeta, str, int]:
+        """Makes TokenMeta data for substring suffixes.
+
+        Args:
+            substring: The `substring` to tokenize.
+            pos: The pointer to the start position of substring in the text.
+
+        Returns:
+            token_meta: The TokenMeta object of the suffix.
+            substring: The updated substring after removing the suffix.
+            pos: The pointer to the start position of new `substring` in the text.
+        """
+        suff_len = self.find_suffix(substring)
```

Comment:
Please add an empty line before this line, after the docstring, and comment the line briefly